test/index.js

```diff
@@ -977,6 +977,24 @@ describe('def', function() {
     eq(dec(new Number(-1)), -2);
   });
 
+  it('supports the "RegexFlags" type', function() {
+    eq($.RegexFlags.name, 'sanctuary-def/RegexFlags');
+    eq($.RegexFlags.test(''), true);
+    eq($.RegexFlags.test('g'), true);
+    eq($.RegexFlags.test('i'), true);
+    eq($.RegexFlags.test('m'), true);
+    eq($.RegexFlags.test('gi'), true);
+    eq($.RegexFlags.test('gm'), true);
+    eq($.RegexFlags.test('im'), true);
+    eq($.RegexFlags.test('gim'), true);
+    //  String objects are not acceptable.
+    eq($.RegexFlags.test(new String('')), false);
+    //  Flags must be alphabetically ordered.
+    eq($.RegexFlags.test('mg'), false);
```

Comment:
I think it would be surprising to users that flags required a certain order. I would find it a bit strange if `grep -rin` got mad at me, or `'xi XI'.replace(/xi/ig, 'hello')` threw